src/components/Popup.js

```diff
@@ -2,45 +2,39 @@ class Popup {
   constructor({ popupSelector }) {
     this._popupElement = document.querySelector(popupSelector);
     this._handleEscClose = this._handleEscClose.bind(this);
+    this._closeButton = this._popupElement.querySelector(
+      ".modal__close-button"
+    );
   }
 
   open() {
     this._popupElement.classList.add("modal__open");
-    this.addEventListeners();
+    document.addEventListener("keydown", this._handleEscClose);
   }
 
   close() {
     this._popupElement.classList.remove("modal__open");
-    this.removeEventListeners();
-  }
-
-  _handleClickClose = (evt) => {
-    if (evt.target === evt.currentTarget) {
+    document.removeEventListener("keydown", this._handleEscClose);
+    this._closeButton.removeEventListener("click", () => {
       this.close();
-    }
```

Comment:
```js
    this._closeButton.removeEventListener("click", () => {

```


Please, delete it . You don't need to remove the close icon listener (and you can't remove it like that anyway


In `close` there will be only 2 lines of code

```js

  close() {
    this._popupElement.classList.remove("modal__open");
    document.removeEventListener("keydown", this._handleEscClose);
  }

```